### ISSUE TITLE
Linux battery capacity reporting 0%

### ISSUE THREAD

**Issue by distek** [alexkainzinger/resource-watchdog#6]
Hi there!

Not sure when this changed, might be related to the upstream `systeminformation` library, but I noticed recently that the remaining percentage on my laptop is always showing AC connected with 0 percent.

Running Arch linux on the latest LTS kernel:

```
$ uname -a
Linux quark 6.1.39-1-lts #1 SMP PREEMPT_DYNAMIC Wed, 19 Jul 2023 17:58:31 +0000 x86_64 GNU/Linux
```

Fresh downloaded version of Code:

```
$ code --version
1.80.2
2ccd690cbff1569e4a83d7c43d45101f817401dc
x64
```

The following diff fixes it for me. Doesn't rely on the `percent` property returned by `systeminformation`, which seems to just be returning `0`:

```diff
diff --git a/src/monitors/battery.ts b/src/monitors/battery.ts
index b7b64ff..9d9284c 100644
--- a/src/monitors/battery.ts
+++ b/src/monitors/battery.ts
@@ -25,8 +25,9 @@ export class Battery extends AbstractResource {
   }

   protected async getDisplay() {
-    const { percent } = await battery();
-    const percentRemaining = Math.min(Math.max(percent, 0), 100);
+    const currentCap = (await battery()).currentCapacity;
+    const maxCap = (await battery()).maxCapacity;
+    const percentRemaining = Math.floor((currentCap / maxCap) * 100);
     return `$(plug) ${percentRemaining}%`;
   }
 }
```

I am not much of a TS dev, so I'm sure there's a better way to do that.

Let me know if you need any other info!

**Comment by alexkainzinger** [comment]
Hi, thanks for reporting!

I was investigating the code on `systeminformation` [battery.js](https://github.com/sebhildebrandt/systeminformation/blob/master/lib/battery.js#L126) and for my understanding it is just using the Linux power-supply class beneath to check for the POWER_SUPPLY_CAPACITY which returns the percentage level 0...100

I would assume that this is reported incorrectly (which would be unlikely tbh. as it would be a Kernel bug ^^) but otherwise I would not have any idea what the issue could be. 

As a fallback, `systeminformation` also calculates the remaining percantage based on `currentCapacity` and `maxCapacity` like you did in the example.

I anyway update the library and published version `1.3.2` of this extension.. let's see. 

--------------

If you could do me a favor and next time this shows up, could you check one of this file:

- `/sys/class/power_supply/BAT1/uevent`
- `/sys/class/power_supply/BAT0/uevent`

what the `POWER_SUPPLY_CAPACITY` is reporting? thanks :)

**Comment by distek** [comment]
Hey!

Not at my computer right now but I also took a look at their library later on and mine is using BAT0 and the uevent info looks as expected: POWER_SUPPLY_CAPACITY was showing my current battery percentage just fine lol so no idea what's going on! I can play with it later to see if I can figure it out
